### PR TITLE
update example to use winit 0.30

### PR DESCRIPTION
**Pull request by BillyDM**
This fixes a crash when closing the example window.

(This is a new PR in a different branch because vscode was making a huge mess trying to rebase with merge conflicts.)

### CODE REVIEW

File: examples/hello-world.rs [hunk 1]
```diff
@@ -9,158 +9,220 @@ use wgpu::{
...
-        WindowBuilder::new()
+struct WindowState {
+    device: wgpu::Device,
+    queue: wgpu::Queue,
+    surface: wgpu::Surface<'static>,
+    surface_config: SurfaceConfiguration,
+
+    font_system: FontSystem,
+    swash_cache: SwashCache,
+    _cache: glyphon::Cache,
```
Comment (grovesNL) [inline examples/hello-world.rs:29]: It looks like some of these fields might not need to be stored as part of `WindowState` if I'm understanding correctly, could we remove any that we don't need to keep around?

Comment (BillyDM) [reply examples/hello-world.rs]: Alright. The only one I could omit was `_cache: glyphon::Cache`.

**Comment by grovesNL** [comment]
Could you double check that the field order is still needed? I believe there were some changes in the new version of wgpu for surface handling so maybe this isn't necessary anymore. Otherwise everything seems fine to me.

**Comment by BillyDM** [comment]
I just tested it and the ordering is still needed. It segfaults if I move the field to the top of the struct.

**Comment by BillyDM** [comment]
Definitely strange since the window is wrapped in an `Arc`. Oh well, hopefully it gets fixed soon.

**Comment by grovesNL** [comment]
Yeah I asked about it and this isn't a known issue on wgpu's end, so I'll look into creating a smaller test case and report it upstream

**Review by grovesNL — APPROVED** [review]
Thanks!

**Pull request merged**